src/reduxFiles/reducers/authentication/authenticationSlice.js

```diff
@@ -0,0 +1,77 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { loginService, signupService } from "services";
+const localStorageDetails = localStorage.getItem("encodedTokenReceived");
+const initialState = {
+  isUserLoggedIn: localStorageDetails ? true : false,
+  encodedTokenReceived: localStorageDetails ? localStorageDetails : "",
+};
+
+const authenticationSignupFunction = async (userDetails) => {
+  try {
+    const signupResponse = await signupService(userDetails);
+    const { data: dataReceived, status } = signupResponse;
+    if (status === 201) {
+      localStorage.setItem("encodedTokenReceived", dataReceived.encodedToken);
+      localStorage.getItem("encodedTokenReceived");
+      return dataReceived;
+    }
+  } catch (signupErrorReceived) {
+    console.error("ERROR OCCURRED DURING SIGNUP IMPLEMENTATION WITH REDUX");
+    return Promise.error(signupErrorReceived);
+  }
+};
+
+const authenticationLoginFunction = async (userDetails) => {
+  try {
+    const loginResponse = await loginService(userDetails);
+    const { data: dataReceived, status } = loginResponse;
+    if (status === 200) {
+      localStorage.setItem("encodedTokenReceived", dataReceived.encodedToken);
+      localStorage.getItem("encodedTokenReceived");
```

Comment:
No need to get localstorage after setting it.

Reply:
Yes would rectify it